algo/src/main/java/com/hartwig/actin/algo/evaluation/complication/PatternMatcher.java

```diff
@@ -1,11 +1,16 @@
 package com.hartwig.actin.algo.evaluation.complication;
 
 import java.util.List;
+import java.util.Optional;
 import java.util.Set;
-
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import com.hartwig.actin.PatientRecord;
+import com.hartwig.actin.clinical.datamodel.Complication;
+import com.hartwig.actin.clinical.datamodel.ImmutableComplication;
 import org.jetbrains.annotations.NotNull;
 
-final class PatternMatcher {
+public final class PatternMatcher {
```

Comment:
A few comments: 
 - I would put the specific complication functions in a separate class with a more concrete name. PatternMatcher previously was independent from a specific datamodel object, and package-private. Maybe "ComplicationFunctions" would be a better name to hold the new functions?